src/Adapters/Storage/Postgres/PostgresStorageAdapter.js

```diff
@@ -1775,6 +1789,36 @@ function processRegexPattern(s) {
   return literalizeRegexPart(s);
 }
 
+function isStartsWithRegex(value) {
+  if (!value || typeof value !== 'string' || !value.startsWith('^')) {
+    return false;
+  }
+
+  const matches = value.match(/\^\\Q.*\\E/);
+  return !!matches;
+}
+
+function isAllValuesRegexOrNone(values) {
+  if (values.length == 0) {
+    return true;
+  }
+
+  var startsWith = isStartsWithRegex(values[0].$regex);
+  for (var i = 1, length = values.length; i < length; ++i) {
```

Comment:
Interesting looking for-loop you have here. 

`for (let i = 0; i < values.length; i += 1) {}`